site/app/templates/admin/Extensions.twig

```diff
@@ -58,22 +58,47 @@
     }
     $(document).ready(function() {
         flatpickr(document.getElementById("late_calendar"), {
+            plugins: [ShortcutButtonsPlugin(
+                    {
+                        button: [
+                            {
+                                label: "Now"
+                            }, 
+                            {
+                                label: "End of time"
+                            }
+                        ],
+                        label: "or",
+                        onClick: (index, fp) => {
+                            let date;
+                            switch (index) {
+                                case 0: 
+                                    date = new Date();
+                                    break;
+                                case 1: 
+                                    date = new Date("9999-01-01 00:00:00");
+                                    break;
+                            }
+                            fp.setDate(date);
+                        }
+                    }
+                )],
             allowInput: true,
             enableTime: false,
             enableSeconds: false,
             time_24hr: true,
             dateFormat: "Y-m-d",
-            onChange: function(selectedDates, dateStr, instance) {
-                var due_date_value = document.getElementById("due_date").innerHTML.substr(0,10);
+            onChange: function (selectedDates, dateStr, instance) {
```

Comment:
In the extensions page, when I click `Now` or `End of time` in the `Or Pick a New Date` box, the content of `Give the Days of Extensions` is not automatically calculated. I guess it has something to do with this `onChange` function. Needs to fix because the query to update extension dates uses the number in `Give the Days of Extensions` in effect.

That being said, I doubt if we need `now` or `infinity` in the extension page. Both do not make much sense. @bmcutler  what do you think?

Reply:
Thanks @zjxiaohan lets here what Barb has to say about this, perhaps that could be taken off completely for the extensions page.

Reply:
Now does make sense for this page.  Often excused absences are entered after-the-fact.  So now could be exactly what we want in those cases.  infinity makes less sense perhaps, but it does allow the instructor to say -- just turn it in before I submit prepare final/semester grades.  Do we have a potential overflow problem since the date is converted into a number of late days allowed?  We might need to clamp a value to prevent overflow?  I'd prefer to keep the UI standard across the site, so leave it.

Can we share code?  I see the same block of code copied & pasted in  4 places here.  Can we make a function/macro?

Reply:
Thanks Barb for introducing how instructors are using it. I am not sure if there will be overflow, but it makes sense to clamp the value because we store the number of days instead of a timestamp.

Also @Fenn-CS I believe auto-calculating `Given the Days of Extensions` is still a problem (the number does not appear after selecting `now` or `infinity`). Could you verify that in your browser?

Reply:
@zjxiaohan that's true, and I am adding a fix for that.